### ISSUE TITLE
Do not de-aliases requests which are same-origin as the containing document

### ISSUE THREAD

**Issue by gitressa** [uBlockOrigin/uBlock-issues#1062]
### Prerequisites

- [x] I verified that this is not a filter issue
    - Filter issues MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues)
    
- [x] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://old.reddit.com/r/uBlockOrigin/)
    
- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
- I tried to reproduce the issue when...
    - [ ] uBlock Origin is the only extension
    - [ ] uBlock Origin with default lists/settings
    - [ ] using a new, unmodified browser profile
- [x] I am running the latest version of uBlock Origin
- [x] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description
When using advanced mode and visiting a URL with a subdomain like `www` (fx `www.nbc.com`) you have to unblock `www.nbc.com` to allow the page to render somewhat.

What if `www.example.org` was allowed to be loaded by default by `example.org`, by adding a setting, which allows certain subdomains to load globally, by defining a list of allowed subdomains under advanced-settings.html. Something like this?

`allowedSubdomainsGlobally: api, www`

### A specific URL where the issue occurs

`https://www.nbc.com/`

### Steps to Reproduce

1. In Advanced mode, visit `https://www.nbc.com/` and see that the page layout is messed up
2. Open uBlock Origin and allow `www.nbc.com`
3. Reload, and see that the layout improves

### Expected behavior:

Certain sub domains like www could be allowed to be loaded, such as `www.example.org`, if you visit example.org.

### Actual behavior:

Sub domains are not loaded, since the feature doesn't exist, to my knowledge.

### Your environment

* uBlock Origin version: 1.27.4
* Browser Name and version: Firefox 76.0.1 (64-bit)
* Operating System and version: Ubuntu 18.04

**Comment by gwarser** [comment]
Page works fine, I don't see the issue.

**Comment by gitressa** [comment]
Ok, thanks for checking @gwarser. I'll try to wipe my settings, and make a fresh start, to see if it fixes it, or if I can recreate it.

**Comment by gorhill** [comment]
Please find the cause first, just use the logger. I rather have you report what was blocking than be stuck with a vague "uBO malfunctioned we'll never know why".

**Comment by uBlock-user** [comment]
There's no such thing as "advanced mode", instead there's an "advanced user" check.

**Comment by gitressa** [comment]
I have reproduced the situation:
1. Clicked "Reset to default settings ..."
2. Checked "I am an advanced user (required reading)"
3. Followed the directions on the [Blocking mode: hard mode](https://github.com/gorhill/uBlock/wiki/Blocking-mode:-hard-mode#how-to-enable-this-mode) page.
4. Visited https://www.nbc.com/

---

![nbc-not-loaded](https://user-images.githubusercontent.com/3491208/82561720-222b5c00-9b74-11ea-846d-3480db6c16bc.png)

---

<details><summary>Logger output</summary>

```
+0
* * 3p block
--
www.nbc.com
3
image
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/favicon.ico
```
```
+0
www.nbc.com
1
image
https://www.nbc.com/generetic/favicon.ico
```
```
+0
* * 3p block
--
www.nbc.com
3
image
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/images/apple-touch-icon.png
```
```
+0
www.nbc.com
1
image
https://www.nbc.com/generetic/images/apple-touch-icon.png
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/generated/chunks/23.af9188d5deef0a35f30a.js
```
```
+0
www.nbc.com
1
script
https://www.nbc.com/generetic/generated/chunks/23.af9188d5deef0a35f30a.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/generated/chunks/1.a1ff7ef5ebb1ebcbf244.js
```
```
+0
www.nbc.com
1
script
https://www.nbc.com/generetic/generated/chunks/1.a1ff7ef5ebb1ebcbf244.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/generated/chunks/5.408084c74c0a75c77df4.js
```
```
+0
www.nbc.com
1
script
https://www.nbc.com/generetic/generated/chunks/5.408084c74c0a75c77df4.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/generated/chunks/2.292d2ab378fb48f79849.js
```
```
+0
www.nbc.com
1
script
https://www.nbc.com/generetic/generated/chunks/2.292d2ab378fb48f79849.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/generated/chunks/0.d35101b649817b05fbdf.js
```
```
+0
www.nbc.com
1
script
https://www.nbc.com/generetic/generated/chunks/0.d35101b649817b05fbdf.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/generated/generetic.56fd06015d4ac2cae996.js
```
```
+0
www.nbc.com
1
script
https://www.nbc.com/generetic/generated/generetic.56fd06015d4ac2cae996.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/generated/chunks/49.960297b6bc5ccdf411cc.js
```
```
+0
www.nbc.com
1
script
https://www.nbc.com/generetic/generated/chunks/49.960297b6bc5ccdf411cc.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/scripts/nbcd_appmeasurement_2_11.js
```
```
+0
www.nbc.com
1
script
https://www.nbc.com/generetic/scripts/nbcd_appmeasurement_2_11.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://pdk.theplatform.com/5.8.7/pdk/tpPdkController.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://entitlement.auth.adobe.com/entitlement/v4/AccessEnabler.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/scripts/ads.js
```
```
+0
www.nbc.com
1
script
https://www.nbc.com/generetic/scripts/ads.js
```
```
+0
* * 3p block
--
www.nbc.com
3
image
https://cdn.betrad.com/pub/icon1.png
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://jssdkcdns.mparticle.com/js/v2/69dedba1e9714049b35bde9e2f9bf059/mparticle.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/generated/chunks/23.af9188d5deef0a35f30a.js
```
```
+0
www.nbc.com
1
script
https://www.nbc.com/generetic/generated/chunks/23.af9188d5deef0a35f30a.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/generated/chunks/1.a1ff7ef5ebb1ebcbf244.js
```
```
+0
www.nbc.com
1
script
https://www.nbc.com/generetic/generated/chunks/1.a1ff7ef5ebb1ebcbf244.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/generated/chunks/5.408084c74c0a75c77df4.js
```
```
+0
www.nbc.com
1
script
https://www.nbc.com/generetic/generated/chunks/5.408084c74c0a75c77df4.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/generated/chunks/2.292d2ab378fb48f79849.js
```
```
+0
www.nbc.com
1
script
https://www.nbc.com/generetic/generated/chunks/2.292d2ab378fb48f79849.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/generated/chunks/0.d35101b649817b05fbdf.js
```
```
+0
www.nbc.com
1
script
https://www.nbc.com/generetic/generated/chunks/0.d35101b649817b05fbdf.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/generated/generetic.56fd06015d4ac2cae996.js
```
```
+0
www.nbc.com
1
script
https://www.nbc.com/generetic/generated/generetic.56fd06015d4ac2cae996.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/generated/chunks/49.960297b6bc5ccdf411cc.js
```
```
+0
www.nbc.com
1
script
https://www.nbc.com/generetic/generated/chunks/49.960297b6bc5ccdf411cc.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/scripts/nbcd_appmeasurement_2_11.js
```
```
+0
www.nbc.com
1
script
https://www.nbc.com/generetic/scripts/nbcd_appmeasurement_2_11.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://pdk.theplatform.com/5.8.7/pdk/tpPdkController.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://entitlement.auth.adobe.com/entitlement/v4/AccessEnabler.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/scripts/ads.js
```
```
+0
www.nbc.com
1
script
https://www.nbc.com/generetic/scripts/ads.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://tverationalplayer.akamaized.net/player/common/plugin/rpHelper.min.js
```
```
+0
* * 3p block
--
www.nbc.com
3
xhr
https://friendship.nbc.co/v2/graphql?variables=%7B%22app%22:%22nbc%22,%22userId%22:%22-8873681774865413895%22,%22device%22:%22web%22,%22platform%22:%22web%22,%22language%22:%22en%22,%22oneApp%22:true,%22name%22:%22homepage%22,%22type%22:%22PAGE%22,%22timeZone%22:%22America%2FNew_York%22,%22ld%22:true,%22profile%22:[%2200000%22]%7D&extensions=%7B%22persistedQuery%22:%7B%22version%22:1,%22sha256Hash%22:%222f971febd4685aba0de57e2fffe8fffd2fdcd5a00215f02358db18741f864516%22%7D%7D
```
```
+0
##+js(set, canRunAds, true)
www.nbc.com
dom
https://www.nbc.com/
```
```
+0
* * 3p block
--
www.nbc.com
3
image
https://cdn.betrad.com/pub/icon1.png
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://jssdkcdns.mparticle.com/js/v2/69dedba1e9714049b35bde9e2f9bf059/mparticle.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://tverationalplayer.akamaized.net/player/common/plugin/rpHelper.min.js
```
```
+0
* * 3p-script block
--
www.nbc.com
3
script
https://tve-common.nbcuni.com/web/js/nbcuni_hash_config.js
```
```
+0
* * 3p block
--
www.nbc.com
3
css
https://e4204.dscb.akamaiedge.net/
aliasURL=https://www.nbc.com/generetic/generated/generetic.db608ef3008f0820dfef.css
```
```
+0
www.nbc.com
1
css
https://www.nbc.com/generetic/generated/generetic.db608ef3008f0820dfef.css
```
```
+0
www.nbc.com
1
doc
https://www.nbc.com/
```
```
+0
https://www.nbc.com/
```
</details>

**Comment by uBlock-user** [comment]
`nbc.com` is hosted on various CNAMEs, and CNAMEs are now subjected to filtering when you block everything that is 3rd-party in Firefox.

**Comment by gorhill** [comment]
`www.nbc.com` is not blocked in hard mode.

**Comment by gorhill** [comment]
Or are you suggesting that the canonical name if any is same origin as the document then should be allowed by default?

**Comment by Yuki2718** [comment]
Related: https://github.com/uBlockOrigin/uBlock-issues/issues/982
Nothing from `www.nbc.com` is blocked by Hard mode on my end too on Firefox with CleanBrowsing DNS.

**Comment by gitressa** [comment]
Is it a bug then? Let me know if I can supply more info or another approach than I already did in https://github.com/uBlockOrigin/uBlock-issues/issues/1062#issuecomment-632074807.

**Comment by gwarser** [comment]
> Is it a bug then?

No. It's not `WWW.nbc.com` what is blocked, but `https://e4204.dscb.akamaiedge.net/` which is hosting the page on behalf of nbc. And that hoster is third party to nbc, so it's blocked.

**Comment by uBlock-user** [comment]
What you're reporting is by-design behaviour when uBO is ran in hard-mode. Breakage is to be expected.

**Comment by gitressa** [comment]
But, if I manually allow `www.nbc.com` and hit reload, the page layout renders ...

**Comment by uBlock-user** [comment]
> if I manually allow `www.nbc.com` and hit reload, the page layout renders

![Capture](https://user-images.githubusercontent.com/21290713/82606858-6a2e9b00-9bd5-11ea-81e4-7b148eb0c3c2.PNG)

like that ?

**Comment by gitressa** [comment]
Here are before and after images:
### Default, layout not rendered
![no-www](https://user-images.githubusercontent.com/3491208/82607269-d18a2200-9bb8-11ea-9a7e-be8370694d27.png)

### www.nbc.com allowed, page renders
![www-allowed](https://user-images.githubusercontent.com/3491208/82607275-d4851280-9bb8-11ea-8a02-5c6792ae1ba9.png)

Here are a few more web sites with the same behavior:
- https://www.cdc.gov/
- https://www.coronaca.gov/
- https://www.virginia.gov/
- https://www.newscientist.com/

You don't seem to be in [Blocking mode: hard mode](https://github.com/gorhill/uBlock/wiki/Blocking-mode:-hard-mode#how-to-enable-this-mode) ...

```
* * 3p block
* * 3p-script block
* * 3p-frame block
```

**Comment by uBlock-user** [comment]
```
* * 3p-script block
* * 3p-frame block
```

those two are optional and are fallback modes if the user wanted to go back to medium mode, not a requirement.

Click the `+all` button and expand the popup panel, so as to verify whether you're being served via CNAME domains like I'm or not.

**Comment by uBlock-user** [comment]
On a second thought as per the first image, something else is blocking css on nbc.com and you were asked by [gorhill](https://github.com/uBlockOrigin/uBlock-issues/issues/1062#issuecomment-632054778) to post logger screenshots which you haven't done yet, so this a filterlist issue, not a uBO issue as per `https://user-images.githubusercontent.com/3491208/82607269-d18a2200-9bb8-11ea-9a7e-be8370694d27.png`

**Comment by gwarser** [comment]
> But, if I manually allow `www.nbc.com` and hit reload, the page layout renders ... Also, I have observed this on many, many other domains.

"allow" is the key word. 

> allow dynamic filters rules override any existing static and dynamic block filters.

Doc: https://github.com/uBlockOrigin/uBlock-issues/wiki/Dynamic-filtering:-rule-syntax#actions

There will not be second pass filtering with CNAME-resolved domain. 

Also (AFAIK) this will also disable scriptlet injection.

You are using dynamic filtering wrong :)

@uBlock-user works fine for me with green rule, how you did this? https://github.com/uBlockOrigin/uBlock-issues/issues/1062#issuecomment-632347509

And log was attached in https://github.com/uBlockOrigin/uBlock-issues/issues/1062#issuecomment-632074807

**Comment by uBlock-user** [comment]
> how you did this?

Notice the CNAME `edgekey.net` is blocked on side because of * * 3p block ;)

**Comment by gitressa** [comment]
Here are screenshots with expanded list. There does seem to be something going on with e4204.dscb.akamaiedge.net and www.nbc.com. I see what you mean :-)

**default**
![2 1-no-www](https://user-images.githubusercontent.com/3491208/82610187-3a27cd80-9bbe-11ea-859b-7659e268b9b2.png)

**www.nbc.com allowed**
![2 2 www-allowed](https://user-images.githubusercontent.com/3491208/82610237-4f046100-9bbe-11ea-9919-a5705af1daee.png)

**Comment by uBlock-user** [comment]
You didn't scroll down to the end of scroll :(

**Comment by gitressa** [comment]
But that actually leads directly back to my original request: An option of allowing certain subdomains to load for **all** domains, like in this case www.nbc.com, which could have been accomplished by adding a setting for defining globally allowed subdomains, such as `www` with `allowedSubdomainsGlobally: api, www`.

**Comment by uBlock-user** [comment]
You didn't test when only uBO is enabled ? Another blocker extension can also cause damage.

**Comment by gitressa** [comment]
I replicated on another machine with Lubuntu 18.04 and a pristine, defaulted Firefox.

**Comment by gitressa** [comment]
Thanks for your patience and explanations @uBlock-user, @gwarser and @gorhill. I have added this to "My rules", since most of the faulty web sites seem to use `akamaiedge.net` and one `fastly.net`, and now www.nbc.com, www.cdc.gov, www.coronaca.gov, and www.newscientist.com look fine:
```
* akamaiedge.net * allow
* fastly.net * allow
```

![akamaiedge net-allowed](https://user-images.githubusercontent.com/3491208/82612758-eddf8c00-9bc3-11ea-8c20-0c8b609f4521.png)

**Comment by gwarser** [comment]
@gitressa do not use allow/green rules! Use grey/noop. Allow rules also disable static filters.

**Comment by gitressa** [comment]
Thanks for the feedback @gwarser! So this is better?
`* akamaiedge.net * noop`

![akamaiedge-noop](https://user-images.githubusercontent.com/3491208/82613745-647d8900-9bc6-11ea-9f34-5b43670fb232.png)

**Comment by gorhill** [comment]
I am going to remove the ability to create `allow` rules from the UI in next release, too many people are not reading the documentation and hurting themselves in the end. There will be an advanced setting for those who really want to be able to create `allow` rules.

It's not just that they are hurting themselves, they are giving bad examples on how to use dynamic filtering to those who don't know better either.

**Comment by Yuki2718** [comment]
This issue should be closed. `www` is most often a main document which hard mode doesn't filter, so no need to noop or allow. Allowing `api` by default is a terrible idea, this is often used for XHR analytics. If it's about CNAME we already have #982 though it's slightly different.

**Comment by gitressa** [comment]
> www is most often a main document which hard mode doesn't filter, so no need to noop or allow.

This is not correct, as I wrote, it happens more and more. Do all these look great in "Blocking mode: hard mode" on your system?
- https://www.cdc.gov/
- https://www.coronaca.gov/
- https://www.virginia.gov/
- https://www.newscientist.com/

If that's the case then my situation seems to be a bug.

> Allowing api by default is a terrible idea, this is often used for XHR analytics (though it's usually 1p domain too).

This new `allowedSubdomainsGlobally` feature is supposed to be editable, and empty by default. `api` and `www` are merely included as examples.

>If it's about CNAME we already have #982 though it's slightly different.

It might be ... so what would be a better solution here, in terms of configuration?

**Comment by gorhill** [comment]
I decline adding `allowedSubdomainsGlobally`. What is currently happening is expected, canonical names of aliased hostnames is to be treated as 3rd-party, by design. Otherwise you are leaving it to servers to dictate what is 1st- or 3rd-party, possibly for their own benefits and against your own interests. There are warnings about hard and medium modes regarding broken sites.

**Comment by gitressa** [comment]
I get your point @gorhill, and accept your decision. But like I stated, this happens more and more frequently, so there is a clear trend ... and the solution is mostly to allow the `www` sudomain.

It would still be nice to get feedback on how the web sites listed in https://github.com/uBlockOrigin/uBlock-issues/issues/1062#issuecomment-632620791 look, in hard mode in your browsers, to make it clear if this is a bug in Ubuntu+Firefox combination, or it is the general situation.

**Comment by gorhill** [comment]
The site is broken in hard mode, but not because `www.nbc.com` itself is blocked, it is not, but because a 3rd-party CDN is blocked. The 3rd-party happens to be the canonical name for `www.nbc.com`.

I [asked above](https://github.com/uBlockOrigin/uBlock-issues/issues/1062#issuecomment-632080013), I am ready to contemplate the idea of not de-aliasing the hostname of the main document, i.e. `www.nbc.com`, it could be argued that not blocking the canonical name of `www.nbc.com` can be no worst than not blocking `www.nbc.com` itself.

**Comment by Yuki2718** [comment]
@gitressa Note I <strong>separately</strong> mentioned CNAME. Those websites are broken only on Firefox due to CNAME uncloaking and what's required to fix them is NOT to noop www but those 3rd-party CDN like fastly or akamaiedge.

**Comment by uBlock-user** [comment]
> I am ready to contemplate the idea of not de-aliasing the hostname of the main document, i.e. www.nbc.com

Isn't `cnameIgnoreRootDocument` setting exactly for that ?

**Comment by gorhill** [comment]
`cnameIgnoreRootDocument` is to ignore `main_frame` requests.

**Comment by gitressa** [comment]
> ... and what's required to fix them is NOT to noop www but those 3rd-party CDN like fastly or akamaiedge.

Thanks for the feedback @Yuki2718, if I understand you correctly, like this? Or is the syntax something else?
`* akamaiedge.net * noop`

@gorhill:

> I am ready to contemplate the idea of not de-aliasing the hostname of the main document, i.e. www.nbc.com

If the result is that users don't have to maintain an ever increasing list of `* akamaiedge.net * allow`,  `* fastly.net * allow`, etc. I am all for that.

**Comment by gorhill** [comment]
> If the result is that users don't have to maintain an ever increasing list

I doubt this will have such a big effect. If anything I feel like `www.nbc.com` might be the odd one, I think it's more common to have static secondary resources to be loaded from say `static.nbc.com` and would be the one aliased, and which would not be solved by the solution contemplated here.

**Comment by gwarser** [comment]
I think someone asked for this before - if document-A has alias-B then treat alias-B as first party.

**Comment by gorhill** [comment]
This: <https://github.com/uBlockOrigin/uBlock-issues/issues/909>?

**Comment by gitressa** [comment]
> ... I feel like www.nbc.com might be the odd one

I suppose it depends on how you define it, @gorhill. These are a some of the domains where I recently had to allow the `www`-version for the page to render into something legible:

- www.brafton.com
- www.chess.com
- www.dailymail.co.uk
- www.daserste.de
- www.dpd.com
- www.getlucius.com
- www.instagram.com
- www.nrk.no
- www.php.net
- www.ups.com
- www.vanityfair.com
- www.w3schools.com
- www.webcomponents.org
- www.webmproject.org

**Comment by gorhill** [comment]
I am just going to comment on the first one -- this surely applies to all other as well.

As an advanced user, I would expect you open the popup panel and find out why the site is broken:

![a](https://user-images.githubusercontent.com/585534/82671927-a67ded80-9c0d-11ea-8149-2da6d559a50e.png)

The firewall pane gives you enough information to understand why the site is broken: there is a 3rd-party to which `www.brafton.com` is aliased to, so it's just a matter of creating a local _noop_ rule for that 3rd-party. By creating an _allow_ rule for `www.brafton.com`, you are misusing dynamic filtering rules.

**Comment by Yuki2718** [comment]
@gitressa Correct and you don't need to make any new rule for akamaiedge because it's now globally nooped. What need to be nooped by CNAME for various sites include edgekey.net, OTOH fastly is what is likely to be nooped already if you had long been used Hard mode. Also note cloaked requests from a site may vary on what DNS resolver you use.

It seems you're new to Hard mode. The work to maintain rules never ends on Hard mode even without CNAME cloaking unless your browsing habit is visiting only limited sites. There are almost infinite cases such that example.org loads images from static.example.net or assets.example.com. So Hard mode is hard mode, have you expected something else? While Medium mode also doesn't have the exact end for the maintenance work, there is a plateau where you no more need frequent update.

But please understand it is not a support forum here. Anyone being to use Hard mode is expected to understand all these staff.

**Comment by gitressa** [comment]
I feel the actual problem is being side railed by hair splitting over which solution is right or wrong, and not acknowledging that the list of web sites I supplied are not legible without intervention.

I get it: Allowing `www` was the wrong method, I should use `noop` in stead, and thanks for pointing that out. But my point is this: if `www` was allowed by default, all this would not be necessary, and the pages would load. So this proposal still sounds good to me:

> I am ready to contemplate the idea of not de-aliasing the hostname of the main document, i.e. www.nbc.com

Thanks @Yuki2718 for pointing out that the operation is not needed if the third-party needed was akamaiedge and it was `noop`'ed globally. My example was done in a reset'ed Hard mode without `* akamaiedge.net * noop`, which I didn't clarify, sorry.

> The work to maintain rules never ends on Hard mode even without CNAME cloaking unless your browsing habit is visiting only limited sites. There are almost infinite cases such that example.org loads images from static.example.net or assets.example.com. So Hard mode is hard mode, have you expected something else?

I know, the maintenance never stops, I have used uBlock Origin Hard Mode for many years. But it was only recently that I saw a sharp spike in having to intervene for pages to render somewhat, and my (faulty, I now know that) was to allow `www`.

**Comment by uBlock-user** [comment]
A sub-set of a case would be `ghacks.net`. Its resources(script/images/css) etc are hosted on two CNAMEs(`marfeelcdn.com`/`fastly.net`), something to consider when addressing this issue.

**Comment by wong8369** [comment]
I have a similar problem.

Both firefox and chrome have the same settings, same ublock versions and same filters (see below), but page loaded by firefox has some resources blocked.
Filters:
uBlock filters
uBlock filters – Badware risks
uBlock filters – Privacy
uBlock filters – Resource abuse
uBlock filters – Unbreak
EasyList
EasyPrivacy
Malware Domain List
Malware domains
Peter Lowe’s Ad and tracking server list
CHN: CJX's EasyList Lite 611 / 612
CHN: EasyList China (中文
Adblock Warning Removal List
Nano Defender Integration
Nano Filters
Nano Filters - Whitelist

### A specific URL where the issue occurs

```https://www.alidns.com/```

### Steps to Reproduce

1. Open ```https://www.alidns.com/``` in firefox and chrome.
2. Page loaded by chrome has a good style but page loaded by firefox not.
3. Disable ublock in firefox and reload the page, it looks as good as chrome with ublock enabled.

### Expected behavior:

The normal resources should not be blocked, the page should look as good as in chrome.
<img width="964" alt="chrome" src="https://user-images.githubusercontent.com/8777505/84596389-16d80180-ae90-11ea-8f8c-e02b0e80f02c.png">

### Actual behavior:

Some resources are blocked in firefox.
<img width="921" alt="firefox" src="https://user-images.githubusercontent.com/8777505/84596463-7df5b600-ae90-11ea-9579-da3f1264c7b5.png">

### Your environment

* uBlock Origin version: 1.27.10
* Browser Name and version: Firefox 77.0.1
* Operating System and version: macOS 10.14.6

 If I access the missing resource directly, it could be displayed normally, for example ```https://www.alidns.com/public/img/logo.png```

Only in the recorder of ublock for firefox I see some blocked items like
```
||tanx.com^$3p ​--​ www.alidns.com ​3 ​image​ https://na61-na62.wagbridge.alibaba.tanx.com/
​​www.alidns.com​ 1​ image​ https://www.alidns.com/public/img/logo.png
```
<img width="1170" alt="firefox_ublock_detail" src="https://user-images.githubusercontent.com/8777505/84597194-e8105a00-ae94-11ea-8bfb-d7e6e0d0d4c5.png">

It seems that these resources are treated as 3rd-party resources and blocked.

firefox network (ublock disabled):
<img width="1041" alt="firefox_network" src="https://user-images.githubusercontent.com/8777505/84597197-f2caef00-ae94-11ea-94ba-d0b86d106058.png">
<img width="1036" alt="firefox_network2" src="https://user-images.githubusercontent.com/8777505/84597210-fe1e1a80-ae94-11ea-9f5b-da2997ae65af.png">

**Comment by wong8369** [comment]
I find that I'm using ```https://dns.alidns.com/dns-query``` as HTTPS DNS in firefox, if I disabled using HTTPS DNS and then restart Firefox, the problem seems be resolved.

**Comment by Yuki2718** [comment]
@wong8369 That's a filter issue so @uBlock-user already fixed it, but please report in [uAssets](https://github.com/uBlockOrigin/uAssets) and not here next time. Also CNAME issue is dependent on some condition so please include with which DNS (preferably public DNS) you see an issue.

**Comment by Aeriem** [comment]
> I [asked above](https://github.com/uBlockOrigin/uBlock-issues/issues/1062#issuecomment-632080013), I am ready to contemplate the idea of not de-aliasing the hostname of the main document, i.e. `www.nbc.com`, it could be argued that not blocking the canonical name of `www.nbc.com` can be no worst than not blocking `www.nbc.com` itself.

@gorhill Wouldn't it mean 3rd-party trackers aliasing as the hostname of the main document wouldn't be caught? And wouldn't that prevent medium and hard modes from blocking unwanted 3rd-party requests?

**Comment by lukashass** [comment]
I just tried to open a PR for this, but noticed they are enabled for prior contributors only :cry: 

Anyhow, have a look [here](https://github.com/gorhill/uBlock/compare/master...lukashass:skip-uncloak-when-same-hostname-as-document?expand=1). I'm not sure if this should have it's own option in the settings, so I didn't do anything there, yet.

**Comment by JulienPalard** [comment]
Spotted this happening with a simple case (no subdomain involved) with https://www.gendarmerie.interieur.gouv.fr/:

`www.gendarmerie.interieur.gouv.fr` is a CNAME to `cs995.wpc.etacdn.net`:

    $ dig www.gendarmerie.interieur.gouv.fr A  +short
    cs995.wpc.etacdn.net.
    152.199.23.85

So when it tries to load scripts, like `<script src="/gie_assets/build/js/progressive_image/ls.parent-fit.min.js"></script>` uBO de-alias the request and blocks it as 3rd party.

**Comment by sheerun** [comment]
I'd always dealias and just disable 3rd party font blocking if cname of html page is the same as some of resources (e.g. fonts) it loads

**Comment by gwarser** [comment]
Can you check now after https://github.com/gorhill/uBlock/commit/161a175bb0596ed6b7734dd7ff77e85204603e96 how it works for you?

**Comment by gwarser** [comment]
Right now in dev builds: https://github.com/uBlockOrigin/uBlock-issues#ubo-dev-builds

Stable - next month.

**Comment by corobin** [comment]
Hi I just want to clarify this change in behaviour, I tried to post this under the actual commit 161a175bb0596ed6b7734dd7ff77e85204603e96 but the post button there doesn't appear to do anything.

What I understand is: if the network request is to the same origin as the root domain, then cname-decloak is not applied to the network request. the cname can be to any domain, this is not checking whether or not the cname itself is same origin.

so for example, a request made from `example.com` to `cdn.example.com` which is cnamed to `a.b.cloudflare.com` will appear in the dynamic filtering pane only as  `cdn.example.com`, and does not show the `cloudflare` cname.

1. is this correct?

2. if yes, since I created dynamic rules using previous versions, would there be an easy way of checking what dynamic rules are no longer needed because cname-uncloak is no longer being applied?

thanks!
